lib/instrumentation/core/net.js

```diff
@@ -61,27 +61,36 @@ function initialize(agent, net) {
   }
 }
 
-// taken from node master on 2013/10/30
-function normalizeConnectArgs(args) {
-  var options = {}
+// taken from node master on 2017/03/13
+function toNumber(x) {
+  return (x = Number(x)) >= 0 ? x : false
+}
+
+function isPipeName(s) {
+  return typeof s === 'string' && toNumber(s) === false
+}
 
-  function toNumber(x) {
-    return (x = Number(x)) >= 0 ? x : false
+function normalizeConnectArgs(args) {
+  if (args.length === 0) {
+    return [{}, null]
   }
-  if (typeof args[0] === 'object' && args[0] !== null) {
-    // connect(options, [cb])
-    options = args[0]
-  } else if (typeof args[0] === 'string' && toNumber(args[0]) === false) {
-    // connect(path, [cb]);
-    options.path = args[0]
+
+  var arg0 = args[0]
+  var options = {}
+  if (typeof arg0 === 'object' && arg0 !== null) {
+    // (options[...][, cb])
+    options = arg0
+  } else if (isPipeName(arg0)) {
+    // (path[...][, cb])
+    options.path = arg0
   } else {
-    // connect(port, [host], [cb])
-    options.port = args[0]
-    if (typeof args[1] === 'string') {
+    // ([port][, host][...][, cb])
+    options.port = arg0
+    if (args.length > 1 && typeof args[1] === 'string') {
       options.host = args[1]
     }
   }
 
   var cb = args[args.length - 1]
-  return typeof cb === 'function' ? [options, cb] : [options]
+  return (typeof cb === 'function') ? [options, cb] : [options, null]
```

Comment:
Your patch bypasses the Node bug by passing `null` for the callback when no callback is provided instead of nothing. This just happens to be the _one_ thing that `Socket#connect` code does check for: https://github.com/nodejs/node/blob/v7.7.2/lib/net.js#L942.